src/components/data/Scroll.js

```diff
@@ -0,0 +1,42 @@
+import classnames from 'classnames';
+
+import { downcastRef } from '../../util/typing';
+
+/**
+ *
+ * @typedef {import('../../types').PresentationalProps} CommonProps
+ * @typedef {import('preact').JSX.HTMLAttributes<HTMLDivElement>} HTMLDivAttributes
+ *
+ * @typedef ScrollProps
+ * @prop {'raised'|'flat'} [variant='raised'] - Render with scroll-hinting
+ *   shadows ('raised') or without ('flat')
+ */
+
+/**
+ * Render a fluid container that scrolls on overflow.
+ *
+ * @param {CommonProps & ScrollProps & HTMLDivAttributes} props
+ */
+export default function Scroll({
+  children,
+  classes,
+  elementRef,
+
+  variant = 'raised',
+
+  ...htmlAttributes
+}) {
+  return (
+    <div
+      {...htmlAttributes}
+      ref={downcastRef(elementRef)}
+      className={classnames(
+        'min-h-0 h-full w-full overflow-auto',
```

Comment:
This is to prevent a vertical container blowout if the `Scroll` is the immediate child of a `flex` container (which it likely is, if it's used with `ScrollContainer`), consistent with how we handle this situation for horizontal `flex` children. 

https://stackoverflow.com/questions/36230944/prevent-flex-items-from-overflowing-a-container , but in vertical because it's a columnar layout.

Reply:
p.s. I can add a comment.

Reply:
> p.s. I can add a comment.

Please do. The explanation from https://stackoverflow.com/a/66689926/434243 is subtle:

> min-width in a flex context: While the default min-width value is 0 (zero), for flex items it is auto. This can make block elements take up much more space than desired, resulting in overflow.